application/CMakeLists.txt

```diff
@@ -31,169 +191,16 @@ endif()
-endif ()
-
-# exr
-if (F3D_MODULE_EXR)
-  target_compile_definitions(f3d PRIVATE F3D_MODULE_EXR)
-endif ()
-
-# dmon
-if (F3D_MODULE_DMON)
-  target_compile_definitions(f3d PRIVATE F3D_MODULE_DMON)
-endif ()
-
-# tinyfiledialogs
-if (F3D_MODULE_TINYFILEDIALOGS)
-  target_compile_definitions(f3d PRIVATE F3D_MODULE_TINYFILEDIALOGS)
-endif ()
-
-if(VTK_VERSION VERSION_LESS 9.3.20240914)
-  # Headless EGL build
-  if (VTK_OPENGL_HAS_EGL)
-    target_compile_definitions(f3d PRIVATE F3D_HEADLESS_BUILD)
-  endif ()
-endif ()
-
-# F3D_STRICT_BUILD
-list(APPEND f3d_compile_options_private ${f3d_strict_build_compile_options})
-if(F3D_STRICT_BUILD)
-  if(MSVC)
-    target_compile_definitions(f3d PRIVATE _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS _CRT_SECURE_NO_WARNINGS)
-  elseif(APPLE)
-    target_compile_definitions(f3d PRIVATE GL_SILENCE_DEPRECATION)
-  endif()
-endif()
-
-# F3D_TESTING_DISABLE_CATCH_ALL
-if (F3D_TESTING_DISABLE_CATCH_ALL)
-  target_compile_definitions(f3d PRIVATE F3D_TESTING_DISABLE_CATCH_ALL)
-endif ()
-
-# Coverage
-list(APPEND f3d_compile_options_public ${f3d_coverage_compile_options})
-list(APPEND f3d_link_options_public ${f3d_coverage_link_options})
-
-# Sanitizer
-list(APPEND f3d_compile_options_public ${f3d_sanitizer_compile_options})
-list(APPEND f3d_link_options_public ${f3d_sanitizer_link_options})
-
-# MacOS Bundle app
-if(F3D_MACOS_BUNDLE)
-  set_target_properties(f3d PROPERTIES MACOSX_BUNDLE TRUE)
-
-  # Add logo icon
-  set(MACOSX_BUNDLE_ICON_FILE logo.icns)
-  set(f3d_ICON ${F3D_SOURCE_DIR}/resources/logo.icns)
-  set_source_files_properties(${f3d_ICON} PROPERTIES
-    MACOSX_PACKAGE_LOCATION "Resources")
-  target_sources(f3d PRIVATE ${f3d_ICON})
-
-  # Set other MacOS vars
-  set(MACOSX_BUNDLE_EXECUTABLE_NAME ${PROJECT_NAME})
-  set(MACOSX_BUNDLE_INFO_STRING ${PROJECT_DESCRIPTION})
-  set(MACOSX_BUNDLE_GUI_IDENTIFIER "app.f3d.F3D")
-  set(MACOSX_BUNDLE_LONG_VERSION_STRING ${F3D_VERSION})
-  set(MACOSX_BUNDLE_BUNDLE_NAME ${PROJECT_NAME})
-  set(MACOSX_BUNDLE_SHORT_VERSION_STRING ${F3D_VERSION})
-  set(MACOSX_BUNDLE_BUNDLE_VERSION ${F3D_VERSION})
-  set(MACOSX_BUNDLE_COPYRIGHT "Michael Migliore, Mathieu Westphal")
-
-  # Generate MacOS bundle using CMake variable generated by plugins
-  get_property(F3D_MACOS_BUNDLE_XML GLOBAL PROPERTY F3D_MACOS_BUNDLE_XML)
-  configure_file("${F3D_SOURCE_DIR}/resources/BundleInfo.plist.in"
-    "${CMAKE_CURRENT_BINARY_DIR}/BundleInfo.plist")
-  set_target_properties(f3d PROPERTIES MACOSX_BUNDLE_INFO_PLIST "${CMAKE_CURRENT_BINARY_DIR}/BundleInfo.plist")
-endif()
-
-if(APPLE)
-  target_link_libraries(f3d PRIVATE "-framework Cocoa")
-endif()
-
-# Windows executable without console
-if(WIN32)
-  if(F3D_WINDOWS_GUI)
-    set_target_properties(f3d PROPERTIES WIN32_EXECUTABLE TRUE)
-  else()
-    # force usage of wWinMainCRTStartup in console mode for Unicode support
-    list(APPEND f3d_link_options_public "/ENTRY:wWinMainCRTStartup")
```

Comment:
I removed this, it makes the CMake logic more complex and it doesnt matter for the f3d GUI usecase.

Reply:
Why? It's still useful for displaying unicode characters for `f3d-console.exe`.

Reply:
Is that ok if I define it all the time ? I'd rather not use option in the cmake function.